core/src/ibc/client_02/manager.rs

```diff
@@ -72,12 +76,54 @@ impl<'a> Manager<'a> {
         Ok(rlp::decode(&data).expect("Illformed client state stored in DB"))
     }
 
-    pub fn query_consensus_state(&self, id: &str, num: ctypes::BlockNumber) -> Result<ConsensusState, String> {
+    pub fn query_consensus_state(
+        &self,
+        id: IdentifierSlice,
+        num: ctypes::BlockNumber,
+    ) -> Result<ConsensusState, String> {
         let kv_store = self.ctx.get_kv_store();
         if !kv_store.has(&path_consensus_state(id, num)) {
             return Err("Consensus state doesn't exist".to_owned())
         }
         let data = kv_store.get(&path_consensus_state(id, num));
         Ok(rlp::decode(&data).expect("Illformed consensus state stored in DB"))
     }
+
+    /* -------- Various verifiers -------- */
+    /// There are some difference between original ICS specification for these verifiers
+    /// 1. They don't take prefix as an argument: each function should decide it by itself.
+    /// 2. They runs on `Manager`, which has a state DB context.
+    /// 3. They don't take a ClientState: each function retrieves it by itself.
+    /// 4. They all take `id`, which indicates the counterparty chain.
+
+    pub fn verify_connection_state(
+        &self,
+        id: IdentifierSlice,
+        proof_height: BlockNumber,
+        proof: Bytes,
+        connection_identifier: IdentifierSlice,
+        connection_end: &ConnectionEnd,
+    ) -> Result<(), String> {
+        let path = format!("connections/{}", connection_identifier);
+        let client_state = self.query(&id)?;
+        if client_state.raw.number < proof_height {
+            return Err("Invalid proof height".to_owned())
+        }
+        let consensus_state = self.query_consensus_state(&id, proof_height)?;
+        let proof_dec: CommitmentProofCounter = rlp::decode(&proof).map_err(|_| "Illformed proof")?;
```

Comment:
This error message will be advanced to include the name of verify function after I add all of `verify_*()` functions and make some reusable routine out of them.